goagen/gen_client/generator.go

```diff
@@ -149,14 +149,16 @@ func (g *Generator) Generate() (_ []string, err error) {
 		if err != nil {
 			return
 		}
+		arrayToStringTmpl = template.Must(template.New("client").Funcs(funcs).Parse(arrayToStringT))
+	}
+
+	if !g.NoTool {
+		var cliPkg string
 		cliPkg, err = codegen.PackagePath(cliDir)
 		if err != nil {
 			return
 		}
```

Comment:
GitHub's diff doesn't make this clear, but all I've done is moved the above `cliPkg` code down to inside the `if !g.NoTool` block.  `cliDir` is only set when `g.NoTool` is false, so using `--notool` would always result in a panic.